### DISCUSSION TITLE
Setting up Sass for development and production

### DISCUSSION THREAD

**Discussion by mcamprecios** [wordplate/wordplate#357] [General]
Hi! Thanks for the boilerplate, it is **great**.

I wanted to use it with Sass instead of vanilla CSS. And I'm trying to understand how Vite and the theme should handle it. It's my first project with Vite, so it may seem a bit stupid. Sorry!

Besides installing sass according to Vite's documentation, I assumed I had to rename the `recources/css/app.css` to `app.scss`. I got lost, but finally I sorted it out. 

To work locally, I changed `functions.php`: 

`wp_enqueue_style('app', 'http://localhost:5173/resources/css/app.css', [], null);` to `app.scss`.

And to build the assets I changed `vite.config.js`: 

```
rollupOptions: {
   input: [
      // 'resources/css/app.css',
      'resources/css/app.scss',
      'resources/js/app.js',
   ]
},
```

Is this the intended way of using Sass? Reading the docs I thought it all hanged from Vite and the theme required no changes. And it seems a bit odd to change it in both the theme `functions.php` file and `vite.config.js`.

Am I doing something wrong?

**A small one:** And if this is the intended way, there is a secondary idea about file naming. Some JS packages that import CSS styles like Swiper.js result in the creation of an extra CSS file. Having that in mind, I find it useful to name the files: `styles.css`, `scripts.js` and `scripts.css`. I may have get used to it because working on my own build process that also uses esbuild, and it outputs both CSS and JS files coming from `whatever.js` with the same name.

Again, I'm not that experienced, so sorry if I'm missing the big picture.

And massive thanks!

**Comment by vinkla** [comment]
I'm not quite sure about the recommended Sass setup. Could it be that this is the appropriate approach for implementing it with Laravel and Vite.js as well?

**Comment by mcamprecios** [comment]
I'm not sure either, when I worked with Laravel we didn't use Vite.js but Webpack.

I may be overthinking it all. May be useful for the next one, but feel free to delete the thread.

**Comment by mcamprecios** [comment]
Just in case it may be useful, I wrote down my settings in my copy of Wordplate: https://gitlab.com/kram08980/wp-starter#css

Thanks again!
